fluent/client/ws_client_test.go

```diff
@@ -303,6 +305,42 @@ var _ = Describe("WSClient", func() {
 			Expect(bytes.Equal(bits, writtenbits)).To(BeTrue())
 		})
 
+		It("Sends the message", func() {
+			msgBytes, _ := msg.MarshalMsg(nil)
+			Expect(client.Send(&msg)).ToNot(HaveOccurred())
+
+			writtenBytes := conn.WriteArgsForCall(0)
+			Expect(bytes.Equal(msgBytes, writtenBytes)).To(BeTrue())
+		})
+
+		When("The message is large", func() {
+			var (
+				seededRand *rand.Rand
+			)
+			const charset = "abcdefghijklmnopqrstuvwxyz" + "ABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+			const messageSize = 65536
+
+			JustBeforeEach(func() {
+				seededRand = rand.New(
+					rand.NewSource(time.Now().UnixNano()))
+				b := make([]byte, messageSize)
+				for i := range b {
+					b[i] = charset[seededRand.Intn(len(charset))]
+				}
+				msg.Record = b
+			})
+
+			It("Sends the correct number of bits", func() {
+				Expect(client.Send(&msg)).ToNot(HaveOccurred())
+				var b bytes.Buffer
+				Expect(msgp.Encode(&b, &msg)).ToNot(HaveOccurred())
+				Expect(conn.WriteCallCount()).NotTo(Equal(0))
+				expectedBytes := len(b.Bytes())
+				writtenBytes := len(conn.WriteArgsForCall(0))
+				Expect(writtenBytes).To(Equal(expectedBytes))
+			})
```

Comment:
What are we supposed to test? From my point of view it's mixed with preparation code.
Encoding into `bytes.Buffer` is NOT subject of the test - so it should not reside in the `It`-block